File: problem-sets/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Prompt the user for change owed, in cents
    int cents;
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);

    // Initialize variables to count the number of coins
    int coins = 0;

    // Calculate the number of quarters needed
    int quarters = cents / 25;
    coins += quarters;
    cents %= 25;

    // Calculate the number of dimes needed
    int dimes = cents / 10;
    coins += dimes;
    cents %= 10;

    // Calculate the number of nickels needed
    int nickels = cents / 5;
    coins += nickels;
    cents %= 5;

    // Pennies remaining are the number of pennies needed
    coins += cents;

    // Print the minimum number of coins needed
    printf("%d\n", coins);

    return 0;
}

File: problem-sets/sentimental-cash/cash.py
from cs50 import get_float

def main():
    # Prompt user for the amount of change owed
    while True:
        dollars = get_float("Change owed: ")
        if dollars >= 0:
            break

    # Convert dollars to cents
    cents = round(dollars * 100)

    # Initialize the number of coins
    coins = 0

    # Calculate the number of quarters
    coins += cents // 25
    cents %= 25

    # Calculate the number of dimes
    coins += cents // 10
    cents %= 10

    # Calculate the number of nickels
    coins += cents // 5
    cents %= 5

    # Calculate the number of pennies
    coins += cents

    # Print the total number of coins
    print(coins)

if __name__ == "__main__":
    main()
